scarb/src/ops/proc_macro_server/methods/defined_macros.rs

```diff
@@ -13,37 +15,52 @@ impl Handler for DefinedMacros {
         proc_macro_host: Arc<ProcMacroHost>,
         _params: Self::Params,
     ) -> Result<Self::Response> {
-        let mut response = proc_macro_host
+        let crate_macro_info = proc_macro_host
             .macros()
             .iter()
-            .map(|e| DefinedMacrosResponse {
-                attributes: e.declared_attributes(),
-                inline_macros: e.inline_macros(),
-                derives: e.declared_derives(),
-                executables: e.executable_attributes(),
-            })
-            .reduce(|mut acc, defined_macros| {
-                acc.attributes.extend(defined_macros.attributes);
-                acc.inline_macros.extend(defined_macros.inline_macros);
-                acc.derives.extend(defined_macros.derives);
-                acc.executables.extend(defined_macros.executables);
+            .map(|macro_instance| {
+                let attributes = macro_instance
+                    .declared_attributes()
+                    .into_iter()
+                    .sorted()
```

Comment:
```suggestion
                    .sorted()
```
With macros per crate there should be no doubled macros. But not sure if this can be done here or later.